### PR TITLE
tiptop: new port

### PR DESCRIPTION
**Pull request by herbygillot**
#### Description

New port for [tiptop](https://github.com/nschloe/tiptop), a system monitoring CLI tool.

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [ ] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL? 
- [x] checked your Portfile with `port lint`?
- [ ] tried existing tests with `sudo port test`?
- [x] tried a full install with `sudo port -d install`?
- [x] tested basic functionality of all binary files?

### CODE REVIEW

File: sysutils/tiptop/Portfile [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+maintainers         {gmail.com:herby.gillot @herbygillot} openmaintainer
+
+checksums           rmd160  a4ae261c51af8ebb5b736c6c2cee39a8658837ad \
+                    sha256  73f8a4367f05acc90a118554bfd5bfd9b645a30b1de4985740b1cf6a1a6486c2 \
+                    size    14932
+
+python.default_version      39
+python.pep517               yes
+
+depends_lib-append  port:py${python.version}-setuptools
```
Comment (reneeotten) [inline sysutils/tiptop/Portfile:31]: shouldn't this be just a `depends_build`? Since you're using Python 3.9 it's not a runtime dependency anymore for the `console_script` since `importlib_metadata` is part of the Python core (3.8+)

File: sysutils/tiptop/Portfile [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+github.tarball_from archive
+revision            0
+
+description         \
+    Command-line system monitoring
+
+long_description    \
+    tiptop is a command-line system monitoring tool in the spirit of top. It \
+    displays various interesting system stats, graphs it, and works on Linux \
+    and macOS.
```
Comment (reneeotten) [inline sysutils/tiptop/Portfile:17]: can this not be just aligned with the rest of the Portfile (I know.. it's a personal preference)

File: sysutils/tiptop/Portfile [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+                    port:py${python.version}-distro \
+                    port:py${python.version}-importlib-metadata \
+                    port:py${python.version}-psutil \
+                    port:py${python.version}-textual
+
+test.run            yes
+test.env-append     PYTHONPATH=./build/lib
+test.cmd            ${python.bin}
+test.pre_args       -c \"import tiptop\; tiptop.run()\"
+test.args           --version
```
Comment (reneeotten) [inline sysutils/tiptop/Portfile:43]: it looks like there are actual tests in the GitHub repository, can these be run instead of just importing it and returning a version?

File: sysutils/tiptop/Portfile (hunk 2 above)
Comment (herbygillot) [reply sysutils/tiptop/Portfile:17]: Definitely personal preference.  I use to use the same alignment for long blocks of text, but I prefer this style nowadays as it feels more readable (to me).

File: sysutils/tiptop/Portfile (hunk 1 above)
Comment (herbygillot) [reply sysutils/tiptop/Portfile]: Change made.

File: sysutils/tiptop/Portfile (hunk 3 above)
Comment (herbygillot) [reply sysutils/tiptop/Portfile]: Done.

**Review by mascguy — APPROVED** [review]

**Comment by mascguy** [comment]
@reneeotten it looks like Herby has addressed the various concerns raised.

Are you fine with merging?

File: sysutils/tiptop/Portfile [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+python.default_version      39
+python.pep517               yes
+
+depends_build-append    \
+                    port:py${python.version}-setuptools \
+                    port:py${python.version}-wheel
+
+depends_run-append  port:py${python.version}-cpuinfo \
+                    port:py${python.version}-distro \
+                    port:py${python.version}-importlib-metadata \
```
Comment (reneeotten) [inline sysutils/tiptop/Portfile:37]: this is only needed for python < 3.8 so it should be removed here

File: sysutils/tiptop/Portfile [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+                    port:py${python.version}-psutil \
+                    port:py${python.version}-textual
+
+depends_test-append port:py${python.version}-pytest
+
+test.run            yes
+test.env-append     PYTHONPATH=./build/lib
+test.cmd            pytest-${python.branch}
+test.pre_args       {}
```
Comment (reneeotten) [inline sysutils/tiptop/Portfile:46]: why is this line needed - typically one sets `test.target` to an empty string. I've never needed to clear `test.pre_arg` before...

File: sysutils/tiptop/Portfile (hunk 2 above)
Comment (reneeotten) [reply sysutils/tiptop/Portfile:17]: I can see that point for the `long_description` but see no need to have the `description` start on a newline

File: sysutils/tiptop/Portfile (hunk 5 above)
Comment (herbygillot) [reply sysutils/tiptop/Portfile:46]: `test.pre_args` is defaulting to the value `test`:
https://github.com/macports/macports-base/blob/999867a094363e006d14bd714be9558933b526be/src/port1.0/porttest.tcl#L23-L24

If not cleared, this happens:
```
Executing:  cd "/opt/macports-test/var/macports/build/_Users_herby_Source_macports-ports_sysutils_tiptop/tiptop/work/tiptop-0.0.16" && pytest-3.9 test
DEBUG: system:  cd "/opt/macports-test/var/macports/build/_Users_herby_Source_macports-ports_sysutils_tiptop/tiptop/work/tiptop-0.0.16" && pytest-3.9 test
ERROR: file or directory not found: test

================================================================================================================ test session starts ================================================================================================================
platform darwin -- Python 3.9.9, pytest-6.2.5, py-1.10.0, pluggy-1.0.0
rootdir: /opt/macports-test/var/macports/build/_Users_herby_Source_macports-ports_sysutils_tiptop/tiptop/work/tiptop-0.0.16
collected 0 items

=============================================================================================================== no tests ran in 0.00s ===============================================================================================================
Command failed:  cd "/opt/macports-test/var/macports/build/_Users_herby_Source_macports-ports_sysutils_tiptop/tiptop/work/tiptop-0.0.16" && pytest-3.9 test
Exit code: 4
```

File: sysutils/tiptop/Portfile (hunk 4 above)
Comment (herbygillot) [reply sysutils/tiptop/Portfile]: Done

File: sysutils/tiptop/Portfile (hunk 5 above)
Comment (reneeotten) [reply sysutils/tiptop/Portfile:45]: okay, so then please do the same as in pretty much every other Python port (and I suggested above) and add the line
```test.target```

Comment (herbygillot) [reply sysutils/tiptop/Portfile]: Changes made.

**Review by reneeotten — APPROVED** [review]

**Comment by mascguy** [comment]
Thanks Renee!
